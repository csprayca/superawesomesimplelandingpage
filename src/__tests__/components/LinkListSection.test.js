/* eslint-env jest */
'use strict';

const React = require('react');
const renderer = require('react-test-renderer');
const LinkListSection = require('../../components/LinkListSection');

const links = [
  {
    title: 'Watch my latest video',
    toURL: 'https://youtu.be/bwESQ8wOJ5Y',
    fromURL: '/latestvideo'
  },
  {
    title: 'Follow behind the scenes on instagram',
    toURL: 'https://www.instagram.com/lastminutelaura.ca',
    fromURL: '/instagram'
  },
  {
    title: 'Checkout my etsy store for one of a kind items',
    toURL: 'https://www.etsy.com/ca/shop/LastMinuteLaura',
    fromURL: '/etsy'
  },
  {
    title: 'Recent Livestream',
    toURL: 'https://youtu.be/ZaSOYhQUQSw',
    fromURL: '/recentlivestream'
  },
  {
    title: 'Youtube',
    toURL: 'https://www.youtube.com/channel/UCB5mFx3KjuOuMZoXDLSUvSg',
    fromURL: '/youtube'
  }
];

describe('LinkListSection component', () => {
  describe('given no input, default settings', () => {
    test('it works as expected', () => {
      const component = renderer.create(<LinkListSection />);
      const renderTree = component.toJSON();

      expect(renderTree).toMatchSnapshot();
    });
  });

  describe('given expected input', () => {
    test('it renders the list as expected', () => {
      const component = renderer.create(<LinkListSection links={links} />);
      const renderTree = component.toJSON();

      expect(renderTree).toMatchSnapshot();
    });
  });
});
