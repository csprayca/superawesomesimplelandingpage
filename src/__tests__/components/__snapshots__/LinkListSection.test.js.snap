// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkListSection component given expected input it renders the list as expected 1`] = `
<section
  className="jsx-79328968"
>
  <ul
    className="jsx-79328968"
  >
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/latestvideo"
        role="button"
        title="Watch my latest video"
      >
        Watch my latest video
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/instagram"
        role="button"
        title="Follow behind the scenes on instagram"
      >
        Follow behind the scenes on instagram
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/etsy"
        role="button"
        title="Checkout my etsy store for one of a kind items"
      >
        Checkout my etsy store for one of a kind items
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/recentlivestream"
        role="button"
        title="Recent Livestream"
      >
        Recent Livestream
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/youtube"
        role="button"
        title="Youtube"
      >
        Youtube
      </a>
    </li>
  </ul>
</section>
`;

exports[`LinkListSection component given no input, default settings it works as expected 1`] = `
<section
  className="jsx-97080905"
>
  <ul
    className="jsx-97080905"
  >
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/latestvideo"
        role="button"
        title="Watch my latest video"
      >
        Watch my latest video
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/instagram"
        role="button"
        title="Follow behind the scenes on instagram"
      >
        Follow behind the scenes on instagram
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/cspray"
        role="button"
        title="Checkout C Spray - a product we made"
      >
        Checkout C Spray - a product we made
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/etsy"
        role="button"
        title="Checkout my etsy store for one of a kind items"
      >
        Checkout my etsy store for one of a kind items
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/recentlivestream"
        role="button"
        title="Recent Livestream"
      >
        Recent Livestream
      </a>
    </li>
    <li
      className="jsx-2036342278"
    >
      <a
        className="jsx-2004290176"
        href="/youtube"
        role="button"
        title="Youtube"
      >
        Youtube
      </a>
    </li>
  </ul>
</section>
`;
